lib/wallet/src/Cardano/Wallet/Primitive/AddressDerivation/Shelley.hs

```diff
@@ -287,70 +289,63 @@ instance WalletKey ShelleyKey where
 instance GetPurpose ShelleyKey where
     getPurpose = purposeCIP1852
 
-instance PaymentAddress 'Mainnet ShelleyKey 'CredFromKeyK where
-    paymentAddress paymentK = do
+instance PaymentAddress ShelleyKey 'CredFromKeyK where
+    paymentAddress SMainnet paymentK = do
```

Comment:
The code is as correct as the previous version. 👍  The diff is a big mangled due to duplication, though.

Could you do a follow-up pull request where you group the case distinction on the networkId into a single definition of `networkTag`? I.e.

```hs
    paymentAddress SMainnet paymentK = do
        Address $ BL.toStrict $ runPut $ do
            putWord8 (enterprise + networkTag)
            putByteString . blake2b224 . xpubPublicKey . getKey $ paymentK
      where
        enterprise = 96
        networkTag = case networkId of
            SMainnet -> 1
            STestnet _ -> 0
```

(To reduce code duplication, it would also make sense to define `paymentAddress` in terms of `liftPaymentAddress`, but let's not go overboard — at some point, we may want to switch to `cardano-addresses` instead.)

Reply:
(Same for the other classes e.g. `delegationAddress`)

Reply:
yes